src/test/java/com/github/movimentocodar/maquinadecafe/maquinadecafe/MaquinaCafeTest.java

```diff
@@ -0,0 +1,35 @@
+package com.github.movimentocodar.maquinadecafe.maquinadecafe;
+
+import com.github.movimentocodar.maquinadecafe.maquinadecafe.bebidas.receitas.AguaQuente;
+import com.github.movimentocodar.maquinadecafe.maquinadecafe.bebidas.receitas.Cafe;
+import org.junit.jupiter.api.Test;
+
+import static org.junit.jupiter.api.Assertions.*;
+
+class MaquinaCafeTest {
+    MaquinaCafe maquinadecafe = new MaquinaCafe();
+
+    @Test
+    void prepararBebidaMaquinaTest() throws Exception {
+        boolean preparar = maquinadecafe.comprarBebida(new AguaQuente());
+        assertEquals(true, preparar);
+    }
+
+    @Test
+    void prepararBebidaJogarExeceptionCasoNaoTenhaAguaMaquinaTest() throws Exception {
+        Exception e = assertThrows(Exception.class , () -> {
```

Comment:
Não costuma ser um bom sinal ter lógica no teste, nesse caso uma estrutura de repetição `for`. Ao invés disso, vc poderia mockar o comportamento do Reservatório para não ter água